### ISSUE TITLE
Traefik 2.0.4 ingress tcp 404

### ISSUE THREAD

**Issue by ekjuanrejon** [containous/traefik#5791]
I have successfully deployed traefik 2.0.4 and it is working in kubernetes

I am having a hard time to get to use traefik with ingress route tcp. 

I get a Service and everything. 

However the tcp service is not working. It has a red exclamation mark

![image](https://user-images.githubusercontent.com/38057696/68317015-4cbf3700-00b2-11ea-8e4b-92ce49da0f46.png)

```json
{
	"routers": {
		"traefik2-traefik2-dashboard-6ea17d312c4cc7fcaaac@kubernetescrd": {
			"entryPoints": ["websecure"],
			"service": "traefik2-traefik2-dashboard-6ea17d312c4cc7fcaaac",
			"rule": "HOST(`xxx.xxxx-xxx.com`)",
			"tls": {
				"certResolver": "xxx"
			},
			"status": "enabled",
			"using": ["websecure"]
		}
	},
	"services": {
		"traefik2-traefik2-dashboard-6ea17d312c4cc7fcaaac@kubernetescrd": {
			"loadBalancer": {
				"servers": [{
					"url": "http://10.244.3.43:9000"
				}],
				"passHostHeader": true
			},
			"status": "enabled",
			"usedBy": ["traefik2-traefik2-dashboard-6ea17d312c4cc7fcaaac@kubernetescrd"],
			"serverStatus": {
				"http://10.244.3.43:9000": "UP"
			}
		}
	},
	"tcpRouters": {
		"couchbase-syncgateway-dev-syncgateway-external-49ed1f12411af8cb3969@kubernetescrd": {
			"entryPoints": ["syncgateway"],
			"service": "couchbase-syncgateway-dev-syncgateway-external-49ed1f12411af8cb3969",
			"rule": "HostSNI(`xxx.xx.xx-xx.com`)",
			"tls": {
				"passthrough": false,
				"certResolver": "xxxx"
			},
			"status": "enabled",
			"using": ["syncgateway"]
		}
	},
	"tcpServices": {
		"couchbase-syncgateway-dev-syncgateway-external-49ed1f12411af8cb3969@kubernetescrd": {
			"loadBalancer": {
				"terminationDelay": 100,
				"servers": [{
					"address": "10.244.3.42:4985"
				}]
			},
			"status": "enabled",
			"usedBy": ["couchbase-syncgateway-dev-syncgateway-external-49ed1f12411af8cb3969@kubernetescrd"]
		}
	}
}
```

**Comment by traefiker** [comment]
Hi! I'm Træfiker :robot: the bot in charge of communication regulation.

Thanks for your interest in Traefik!

Issue templates help us help you by providing all necessary information.

Please edit your issue and use the available templates:

- [Feature request](https://raw.githubusercontent.com/containous/traefik/master/.github/ISSUE_TEMPLATE/Feature_request.md)
- [Bug report](https://raw.githubusercontent.com/containous/traefik/master/.github/ISSUE_TEMPLATE/Bug_report.md)

And remember: each time someone ignores the template, a cute little bunny dies.
<details>
<summary>#SaveTheCuteBunny :heart: :rabbit: :heart:</summary>

![southpark - estrella kill rabbit](https://user-images.githubusercontent.com/5674651/27143914-3cea7896-5130-11e7-81f8-02d8ac2da7a7.gif)

![southpark - estrella kill rabbit2](https://user-images.githubusercontent.com/5674651/27143924-49104c72-5130-11e7-82ae-ccedbd5dfa1c.gif)

</details>

**Comment by JavierPandu** [comment]
I'm having a similar issue:
`- "traefik.tcp.routers.sograce.rule=HostSNI(`*`)"
        - "traefik.tcp.routers.myrouter.entrypoints=mytcp"
        - "traefik.tcp.routers.myrouter.service=myservice@docker"        
        - "traefik.tcp.services.myservice.loadbalancer.server.port=9992" `
The service is up and from within I can telnet the port. But I get connection refused when going through traefik.
The traefik service does have the port 9992:9992 mapped
The exclamation mark is there in the service, everything else seems fine

**Comment by cbwsports** [comment]
@ekjuanrejon Did you ever figure out what was wrong? I am experiencing the same thing at the moment. Although the router is pointing to the correct server there is a red exclamation mark and I am not able to use tcp. Please update with your solution if you found one. Thanks

**Comment by dmaasland** [comment]
I'm having this exact problem. My TCP service is showing an exclamation mark next to my server. Funny thing is, everything is working fine. These are my labels:

```yaml
    [..]
    labels:
      ## Traefik
      #  Enable
      - "traefik.enable=true"
      #  Router
      - "traefik.tcp.routers.esmc-server.rule=HostSNI(`*`)"
      - "traefik.tcp.routers.esmc-server.entrypoints=eset-agent"
      - "traefik.tcp.routers.esmc-server.service=esmc-server"
      - "traefik.tcp.routers.esmc-server.tls=true"
      - "traefik.tcp.routers.esmc-server.tls.passthrough=true"
      #  Service
      - "traefik.tcp.services.esmc-server.loadbalancer.server.port=2222"

  traefik:
    image: traefik:v2.1
    command:
      - "--api.insecure=true"
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
      - "--entrypoints.http.address=:80"
      - "--entrypoints.eset-agent.address=:2222"
    ports:
      - "80:80"
      - "8080:8080"
      - "2222:2222"
```

**Comment by ldez** [comment]
Closed by #6313
